File: frontend/src/types/domain.ts
export type Role = 'admin' | 'judge';

export interface User {
  id: string;
  username: string;
  display_name: string;
  role: Role;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Category {
  id: string;
  name: string;
  description: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Project {
  id: string;
  category_id: string;
  category_name?: string;
  title: string;
  short_description: string;
  full_description: string;
  concept: string;
  designer_name: string;
  team_name: string;
  image_url: string;
  proposal_link: string;
  social_media_link: string;
  drive_link: string;
  attached_file_link: string;
  extra_details: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ScoringCriterion {
  id: string;
  category_id: string;
  name: string;
  description: string;
  max_score: number;
  display_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface VoteScore {
  id: string;
  vote_id: string;
  criterion_id: string;
  score: number;
  created_at: string;
  updated_at: string;
}

export interface Vote {
  id: string;
  judge_id: string;
  project_id: string;
  total_score: number;
  submitted_at?: string | null;
  created_at: string;
  updated_at: string;
  scores: VoteScore[];
}

export interface JudgeProjectCard {
  id: string;
  category_id: string;
  title: string;
  short_description: string;
  designer_name: string;
  team_name: string;
  image_url: string;
  has_voted: boolean;
  current_score?: number | null;
  category_name: string;
}

export interface JudgeProjectDetail {
  project: Project;
  criteria: ScoringCriterion[];
}

export interface JudgeSummaryRow {
  ranking: number;
  project_id: string;
  project_name: string;
  total_score: number;
  has_voted: boolean;
  category_id: string;
}

export interface DashboardStats {
  total_projects: number;
  total_judges: number;
  total_categories: number;
  total_votes_submitted: number;
  completion_percentage: number;
  possible_vote_count: number;
}

export interface AdminProjectRanking {
  ranking: number;
  project_id: string;
  project_name: string;
  category: string;
  total_score: number;
  average_score: number;
  submitted_votes: number;
  completion_percent: number;
}

export interface AdminJudgeVoteRow {
  vote_id: string;
  project_id: string;
  project_name: string;
  category: string;
  judge_id: string;
  judge_name: string;
  total_score: number;
  submitted_at?: string | null;
}

export interface ResultsResponse {
  rankings: AdminProjectRanking[];
  judge_votes: AdminJudgeVoteRow[];
}

export interface ProjectCriterionRow {
  criterion_id: string;
  criterion_name: string;
  max_score: number;
  score: number;
}

export interface ProjectJudgeVoteRow {
  vote_id: string;
  judge_id: string;
  judge_name: string;
  total_score: number;
  submitted_at?: string | null;
  scores: ProjectCriterionRow[];
}

export interface ProjectVoteDetail {
  project: Project;
  combined_score: number;
  judge_votes: ProjectJudgeVoteRow[];
}

export interface LoginResponse {
  token: string;
  user: User;
}

export interface CategoryPayload {
  name: string;
  description: string;
  is_active: boolean;
}

export interface ProjectPayload {
  category_id: string;
  title: string;
  short_description: string;
  full_description: string;
  concept: string;
  designer_name: string;
  team_name: string;
  image_url: string;
  proposal_link: string;
  social_media_link: string;
  drive_link: string;
  attached_file_link: string;
  extra_details: string;
  is_active: boolean;
}

export interface CriterionPayload {
  category_id: string;
  name: string;
  description: string;
  max_score: number;
  display_order: number;
  is_active: boolean;
}

export interface JudgePayload {
  username: string;
  display_name: string;
  password?: string;
  role: Role;
  is_active: boolean;
  category_ids: string[];
}

export interface VoteSubmissionPayload {
  scores: Array<{
    criterion_id: string;
    score: number;
  }>;
}

File: backend/internal/models/domain.go
package models

import "time"

const (
	RoleAdmin = "admin"
	RoleJudge = "judge"
)

type User struct {
	ID            string    `json:"id"`
	Username      string    `json:"username"`
	DisplayName   string    `json:"display_name"`
	PasswordHash  string    `json:"-"`
	Role          string    `json:"role"`
	IsActive      bool      `json:"is_active"`
	ScoredCount   int       `json:"scored_count,omitempty"`
	AssignedCount int       `json:"assigned_count,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type AwardGroup struct {
	ID            string    `json:"id"`
	Code          string    `json:"code"`
	Name          string    `json:"name"`
	NameTh        string    `json:"name_th"`
	Description   string    `json:"description"`
	DescriptionTh string    `json:"description_th"`
	DisplayOrder  int       `json:"display_order"`
	IsActive      bool      `json:"is_active"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// JudgeAwardGroup is the per-judge view of an award group used on the
// group selection screen. Assigned indicates whether the current judge may
// open it; locked groups are shown greyed out.
type JudgeAwardGroup struct {
	ID            string `json:"id"`
	Code          string `json:"code"`
	Name          string `json:"name"`
	NameTh        string `json:"name_th"`
	Description   string `json:"description"`
	DescriptionTh string `json:"description_th"`
	DisplayOrder  int    `json:"display_order"`
	Assigned      bool   `json:"assigned"`
	CategoryCount int    `json:"category_count"`
}

type Category struct {
	ID            string    `json:"id"`
	AwardGroupID  *string   `json:"award_group_id"`
	Name          string    `json:"name"`
	NameTh        string    `json:"name_th"`
	Description   string    `json:"description"`
	DescriptionTh string    `json:"description_th"`
	DisplayOrder  int       `json:"display_order"`
	IsActive      bool      `json:"is_active"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type Project struct {
	ID               string    `json:"id"`
	CategoryID       string    `json:"category_id"`
	CategoryName     string    `json:"category_name,omitempty"`
	Title            string    `json:"title"`
	ShortDescription string    `json:"short_description"`
	FullDescription  string    `json:"full_description"`
	Concept          string    `json:"concept"`
	DesignerName     string    `json:"designer_name"`
	TeamName         string    `json:"team_name"`
	ImageURL         string    `json:"image_url"`
	ImageSourceURL   string    `json:"image_source_url"`
	SocialMediaLink  string    `json:"social_media_link"`
	DriveLink        string    `json:"drive_link"`
	ExtraDetails     string    `json:"extra_details"`
	SpecialDetails   string    `json:"special_details"`
	IsActive         bool      `json:"is_active"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

type ScoringCriterion struct {
	ID            string    `json:"id"`
	CategoryID    string    `json:"category_id"`
	Name          string    `json:"name"`
	NameTh        string    `json:"name_th"`
	Description   string    `json:"description"`
	DescriptionTh string    `json:"description_th"`
	MaxScore      int       `json:"max_score"`
	DisplayOrder  int       `json:"display_order"`
	IsActive      bool      `json:"is_active"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type Vote struct {
	ID          string               `json:"id"`
	JudgeID     string               `json:"judge_id"`
	ProjectID   string               `json:"project_id"`
	TotalScore  int                  `json:"total_score"`
	SubmittedAt *time.Time           `json:"submitted_at"`
	CreatedAt   time.Time            `json:"created_at"`
	UpdatedAt   time.Time            `json:"updated_at"`
	Scores      []VoteCriterionScore `json:"scores,omitempty"`
}

type VoteCriterionScore struct {
	ID          string    `json:"id"`
	VoteID      string    `json:"vote_id"`
	CriterionID string    `json:"criterion_id"`
	Score       int       `json:"score"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type JudgeProjectCard struct {
	ID               string `json:"id"`
	CategoryID       string `json:"category_id"`
	Title            string `json:"title"`
	ShortDescription string `json:"short_description"`
	DesignerName     string `json:"designer_name"`
	TeamName         string `json:"team_name"`
	ImageURL         string `json:"image_url"`
	ImageSourceURL   string `json:"image_source_url"`
	HasVoted         bool   `json:"has_voted"`
	CurrentScore     *int   `json:"current_score"`
	CategoryName     string `json:"category_name"`
}

type JudgeProjectDetail struct {
	Project  Project            `json:"project"`
	Criteria []ScoringCriterion `json:"criteria"`
}

type JudgeSummaryRow struct {
	Ranking     int    `json:"ranking"`
	ProjectID   string `json:"project_id"`
	ProjectName string `json:"project_name"`
	TotalScore  int    `json:"total_score"`
	HasVoted    bool   `json:"has_voted"`
	CategoryID  string `json:"category_id"`
}

type DashboardStats struct {
	TotalProjects        int     `json:"total_projects"`
	TotalJudges          int     `json:"total_judges"`
	TotalCategories      int     `json:"total_categories"`
	TotalVotesSubmitted  int     `json:"total_votes_submitted"`
	CompletionPercentage float64 `json:"completion_percentage"`
	PossibleVoteCount    int     `json:"possible_vote_count"`
}

type LandingStats struct {
	TotalAwards         int `json:"total_awards"`
	TotalActiveProjects int `json:"total_active_projects"`
}

type AdminProjectRanking struct {
	Ranking           int     `json:"ranking"`
	ProjectID         string  `json:"project_id"`
	ProjectName       string  `json:"project_name"`
	CategoryID        string  `json:"category_id"`
	Category          string  `json:"category"`
	TotalScore        int     `json:"total_score"`
	AverageScore      float64 `json:"average_score"`
	SubmittedVotes    int     `json:"submitted_votes"`
	CompletionPercent float64 `json:"completion_percent"`
}

type AdminJudgeVoteRow struct {
	VoteID      string     `json:"vote_id"`
	ProjectID   string     `json:"project_id"`
	ProjectName string     `json:"project_name"`
	Category    string     `json:"category"`
	JudgeID     string     `json:"judge_id"`
	JudgeName   string     `json:"judge_name"`
	TotalScore  int        `json:"total_score"`
	SubmittedAt *time.Time `json:"submitted_at"`
}

type ProjectVoteDetail struct {
	Project       Project               `json:"project"`
	CombinedScore int                   `json:"combined_score"`
	JudgeVotes    []ProjectJudgeVoteRow `json:"judge_votes"`
}

type ProjectJudgeVoteRow struct {
	VoteID      string                `json:"vote_id"`
	JudgeID     string                `json:"judge_id"`
	JudgeName   string                `json:"judge_name"`
	TotalScore  int                   `json:"total_score"`
	SubmittedAt *time.Time            `json:"submitted_at"`
	Scores      []ProjectCriterionRow `json:"scores"`
}

type ProjectCriterionRow struct {
	CriterionID   string `json:"criterion_id"`
	CriterionName string `json:"criterion_name"`
	MaxScore      int    `json:"max_score"`
	Score         int    `json:"score"`
}

type ResultsResponse struct {
	Rankings   []AdminProjectRanking `json:"rankings"`
	JudgeVotes []AdminJudgeVoteRow   `json:"judge_votes"`
}

type CSVExportRow struct {
	Category            string
	ProjectTitle        string
	JudgeName           string
	CriterionName       string
	CriterionScore      int
	ProjectTotalByJudge int
	ProjectTotalOverall int
	AverageScore        float64
}
